lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -207,18 +212,52 @@ protected function _getPortableTableColumnDefinition($tableColumn)
     }
 
     /**
+     * Return Doctrine/Mysql-compatible column default values for MariaDB 10.2.7+ servers.
+     *
+     * - Since MariaDb 10.2.7 column defaults stored in information_schema are now quoted
+     *   to distinguish them from expressions (see MDEV-10134).
+     * - CURRENT_TIMESTAMP, CURRENT_TIME, CURRENT_DATE are stored in information_schema
+     *   as current_timestamp(), currdate(), currtime()
+     * - Note: Quoted 'NULL' is not enforced by Maria, it is technically possible to have
+     *   null in some circumstances (see https://jira.mariadb.org/browse/MDEV-14053)
+     *
+     * @link https://mariadb.com/kb/en/library/information-schema-columns-table/
+     * @link https://jira.mariadb.org/browse/MDEV-13132
+     *
+     * @param null|string $columnDefault default value as stored in information_schema for MariaDB >= 10.2.7
+     */
+    private function getMariaDb1027ColumnDefault(MariaDb1027Platform $platform, ?string $columnDefault) : ?string
```

Comment:
Given that MariaDB is a separate platform, it should have its own schema manager like other platforms do. This logic doesn't belong to the MySQL schema manager. Not sure how feasible this is in this pull request, but it should be done at some point.

Reply:
@morozov, `PostgreSqlSchemaManager` has many plaforms , see `PostgreSQL[91|92|94]Plaform` (same for `SQLServer`, `SQLAnywhere`). More importantly, `MySQL` had already the `DrizzlePlatform` in the same vein.

Following this logic, I think we can save ourselves to introduce a specific schema manager. At least till Maria and Mysql keeps inline. I believe we can consider them a versions of a same product... at least in term of functionnalities used/required by doctrine. Note also, that their fix in the information_schema will probably be implemented in future MySQLs too (if they want to support expressions in defaults, in a sense the new Maria behaviour is now more standard, like Postgres, Oracle...).

Reply:
Let's leave it like this for now. Unlike version-specific platforms which inherit each other, there's no example of version-specific schema managers in the DBAL, so implementing one would require some more discussion.

In general, the fact that something is implemented like it is doesn't mean that it's the best possible approach. The DBAL is almost 10 years old.

Reply:
Fully agree.